server/src/main/java/org/elasticsearch/index/analysis/AbstractIndexAnalyzerProvider.java

```diff
@@ -40,7 +40,10 @@
     public AbstractIndexAnalyzerProvider(IndexSettings indexSettings, String name, Settings settings) {
         super(indexSettings);
         this.name = name;
-        this.version = Analysis.parseAnalysisVersion(this.indexSettings.getSettings(), settings, logger);
+
+        Analysis.deprecateSetting("version", Version.V_8_0_0, indexSettings, settings, name, deprecationLogger);
+        // resolve the analysis version based on the version the index was created with
+        this.version = indexSettings.getIndexVersionCreated().luceneVersion;
```

Comment:
I left version for analyzers because it is passed to lucene and I wasn't sure if it is safe to remove it.

Reply:
++, I don't think we have an official analyzer that uses this information but it might be used in plugins so let's keep it during the deprecation period.